scipy/integrate/tests/test_quadpack.py

```diff
@@ -503,6 +503,27 @@ def f(x, y, z):
             error_tolerance=6e-8
         )
 
+    def test_complex(self):
+        def tfunc(x):
+            return np.exp(1j*x)
+
+        assert_quad(quad(tfunc, 0, np.pi/2, complex_func=True),
+                    1+1j, error_tolerance=1e-6)
+
+        kwargs = {'a': 0, 'b': np.inf, 'full_output':True, 'weight':'cos', 'wvar': 1}
```

Comment:
I'd suggest noting that this is a divergent integral so that `message` and `explain` get returned for the real integral.